src/Common/tests/Performance/PerfTestBase.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.IO;
+
+namespace System
+{
+    /// <summary>Base class for performance test classes</summary>
+    public abstract class PerfTestBase
+    {
+        /// <summary>
+        /// Helper method to create a string containing a number of random
+        /// characters equal to the specified length
+        /// </summary>
+        public static string CreateString(int length)
+        {
+            StringBuilder builder = new StringBuilder();
+            StringBuilder empty = new StringBuilder();
+            while (builder.Length < length)
+            {
+                string toAppend = Guid.NewGuid().ToString();
+                int toAppendLength = Math.Min(toAppend.Length, length - builder.Length);
+                builder.Append(toAppend, 0, toAppendLength);
+            }
+            return builder.ToString();
+        }
+
+        /// <summary>Gets a test file full path that is associated with the call site.</summary>
+        /// <param name="index">An optional index value to use as a suffix on the file name.  Typically a loop index.</param>
+        /// <param name="memberName">The member name of the function calling this method.</param>
+        /// <param name="lineNumber">The line number of the function calling this method.</param>
+        protected string GetTestFilePath(int? index = null, [CallerMemberName] string memberName = null, [CallerLineNumber] int lineNumber = 0)
```

Comment:
They cleanup of the files currently is left to the test using GetTestFilePath and is not handled by PerfTestBase.

I'd prefer not to add the extra overhead in creating a testdirectory for perf tests that didnt use it (of which there are many), Also, I think Eric's suggestion above of modifying the class to be a static PerfUtils would be preferable anyways, as it would allow classes to inherit from other test parents (such as FileCleanupTestBase).